integration_tests/src/fixture_generator/parser.rs

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under both the MIT license found in the
+// LICENSE-MIT file in the root directory of this source tree and the Apache
+// License, Version 2.0 found in the LICENSE-APACHE file in the root directory
+// of this source tree.
+
+//! This module contains the CLI argument definitions and parser.
+
+use akd::storage::types::{AkdLabel, AkdValue};
+use clap::{AppSettings, Parser};
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+
+/// Any alphanumeric string e.g. "User123"
+const USER_PATTERN: &str = r"\w+";
+
+/// A solo string of digits e.g. "10" or a tuple of digits and a string
+/// e.g."(10, 'abc')"
+const EVENT_PATTERN: &str = r"\d+|(\(\s*(\d+)\s*,\s*'(\w*)'\s*\))";
+
+/// An update the tool should include in the tree at the given epoch.
+/// If "value" is None, the tool will randomly generate a value for the epoch.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct UserEvent {
+    pub epoch: u32,
+    pub value: Option<AkdValue>,
+}
+
+/// A user whose update events should be included in the tree.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct User {
+    pub label: AkdLabel,
+    pub events: Vec<UserEvent>,
+}
+
+/// This tool allows a directory to be created with specified and random
+/// contents, capturing the directory state and epoch-to-epoch delta in
+/// an output file for use in debugging and as test fixtures.
+#[derive(Parser, Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[clap(setting = AppSettings::DeriveDisplayOrder)]
+pub struct Args {
+    /// Users and their associated update events.
+    /// A username is expected, followed by a colon and a list of epochs OR
+    /// (epoch, value). Usernames are expected to be utf-8 strings, which will
+    /// be internally interpreted as bytes.
+    /// The following are valid examples of user arguments:
+    ///   --user "username: 1, 3, (5, 'xyz')"
```

Comment:
If the value is not given, is it going to be random?

Reply:
Yes!